src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,115 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elements;
+    private int size;
+
+    public ArrayList(int capacity) {
+        if (capacity <= 0) {
+            throw new IllegalArgumentException("Illegal Capacity: " + capacity);
+        }
+        elements = new Object[capacity];
+    }
+
+    public ArrayList() {
+        this(DEFAULT_CAPACITY);
+    }
+
+    private void checkIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("Index out of bounds: " + index);
+        }
+    }
+
+    private void resizeIfNeed() {
+        if (elements.length == size) {
+            Object[] newArray = new Object[elements.length * 2];
+            System.arraycopy(elements, 0, newArray, 0, size);
+            elements = newArray;
+        }
+    }
+
+    private int indexOf(T element) {
+        for (int i = 0; i < size; i++) {
+            if (element == elements[i] || element != null && element.equals(elements[i])) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
     @Override
     public void add(T value) {
-
+        resizeIfNeed();
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Index out of bounds: " + index);
+        }
+        resizeIfNeed();
+        System.arraycopy(elements, index, elements, index + 1, size - index);
+        elements[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        resizeIfNeed();
+        for (int i = 0; i < list.size(); i++) {
+            elements[size] = list.get(i);
+            size++;
+        }
```

Comment:
if you array has default capacity (10), and you try to add 15 elements what happens? 
you can just reuse the method `add(T value)` inside loop.